electrumsv/daemon.py

```diff
@@ -312,9 +357,23 @@ def run_cmdline(self, config_options: dict) -> Any:
         result = func(*args, **kwargs)
         return result
 
+    def on_stop(self):
+        if self.rest_server.is_alive:
+            app_state.async_.spawn_and_wait(self.rest_server.stop)
+            self.logger.debug("stopped.")
+
     def run(self) -> None:
         while self.is_running():
             self.server.handle_request() if self.server else time.sleep(0.1)
+            if app_state.has_app():
```

Comment:
I do not understand why some startup code is in the main loop.

Reply:
Yes. Will fix along with my earlier comment about when to and when not to spin up the restapi.

Reply:
Fixed